tests/sles4sap/microsoft_sdaf/sdaf_deploy_hanasr.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary:  Executes setup of HanaSR scenario using SDAF ansible playbooks according to:
+#           https://learn.microsoft.com/en-us/azure/sap/automation/tutorial#sap-application-installation
+# Playbooks can be found in SDAF repo: https://github.com/Azure/sap-automation/tree/main/deploy/ansible
+
+# Required OpenQA variables:
+#     'SDAF_ENV_CODE'  Code for SDAF deployment env.
+#     'SDAF_WORKLOAD_VNET_CODE' Virtual network code for workload zone.
+#     'PUBLIC_CLOUD_REGION' SDAF internal code for azure region.
+#     'SAP_SID' SAP system ID.
+#     'SDAF_DEPLOYER_RESOURCE_GROUP' Existing deployer resource group - part of the permanent cloud infrastructure.
+
+# Optional:
+#     'SDAF_ANSIBLE_VERBOSITY_LEVEL' Override default verbosity for 'ansible-playbook'.
+
+use parent 'sles4sap::microsoft_sdaf_basetest';
+use strict;
+use warnings;
+use sles4sap::sdaf_deployment_library;
+use sles4sap::console_redirection;
+use serial_terminal qw(select_serial_terminal);
+use testapi;
+
+sub test_flags {
+    return {fatal => 1};
+}
+
+sub run {
+    serial_console_diag_banner('Module sdaf_deploy_hanasr.pm : start');
+    my $sap_sid = get_required_var('SAP_SID');
+    my $vnet_code = get_required_var('SDAF_WORKLOAD_VNET_CODE');
+
+    connect_target_to_serial();
+    load_os_env_variables();
+
+    sdaf_execute_playbook(playbook_filename => 'pb_get-sshkey.yaml', sap_sid => $sap_sid, vnet_code => $vnet_code, timeout => 90);
+    sdaf_execute_playbook(playbook_filename => 'playbook_00_validate_parameters.yaml', sap_sid => $sap_sid, vnet_code => $vnet_code, timeout => 120);
+    sdaf_execute_playbook(playbook_filename => 'playbook_01_os_base_config.yaml', sap_sid => $sap_sid, vnet_code => $vnet_code);
+    sdaf_execute_playbook(playbook_filename => 'playbook_02_os_sap_specific_config.yaml', sap_sid => $sap_sid, vnet_code => $vnet_code);
+    sdaf_execute_playbook(playbook_filename => 'playbook_04_00_00_db_install.yaml', sap_sid => $sap_sid, vnet_code => $vnet_code);
+    sdaf_execute_playbook(playbook_filename => 'playbook_04_00_01_db_ha.yaml', sap_sid => $sap_sid, vnet_code => $vnet_code);
+    sdaf_execute_playbook(playbook_filename => 'playbook_07_00_00_post_installation.yaml', sap_sid => $sap_sid, vnet_code => $vnet_code);
+    sdaf_execute_playbook(playbook_filename => 'playbook_08_00_00_post_configuration_actions.yaml', sap_sid => $sap_sid, vnet_code => $vnet_code);
```

Comment:
Suggestion:
```
my @playbooks = qw(pb_get-sshkey playbook_00_validate_parameters playbook_01_os_base_config
                   playbook_02_os_sap_specific_config playbook_04_00_00_db_install playbook_04_00_01_db_ha
                   playbook_07_00_00_post_installation playbook_08_00_00_post_configuration_actions);
my %pb_timeout = ( 'pb_get-sshkey' => 90, playbook_00_validate_parameters => 120 );
foreach my $pb (@playbooks) {
    sdaf_execute_playbook(playbook_filename => "$pb.yaml", sap_sid => $sap_sid, vnet_code => $vnet_code, timeout => $pb_timeout{$pb});
}
```

Reply:
Yeah I am itching to add those into a loop, but I would rather keep them as a block for easier readability. I think it might be easier to grasp what's going on in the module just visually without reading the loop.   
If you check conversation with Michele, I am going to remove few arguments to simplify things and I will need to execute a command after running first playbook.

Reply:
Makes sense.